slack_wrapper.gs

```diff
@@ -0,0 +1,258 @@
+    this.args.response_url = metadata_parsed.response_url;
+    
+    this.args.uniqueid = metadata_parsed.uniqueid;
+    this.args.more = par.view.state.values;
+  }
+  
+  
+  parseEventSlashCommand(par){
+    this.token = par.token;
+    this.teamid = par.team_id;
+    
+    this.type = 'command';
+    this.subtype = par.command;
+    
+    this.args.channelid = par.channel_id;
+    this.args.userid = par.user_id;
+    this.args.username = par.user_name;
+    this.args.response_url = par.response_url;
+    this.args.trigger_id = par.trigger_id;
+    
+    this.argsString=par.text;
+    this.parseEventSlashCommandTxt(par.text); // populates this.args
+  }
+  
+  parseEventSlashCommandTxt(txt){
+    // parse txt string and store parsed values in this.args
+    if(txt) {
+      var args = txt.split(' ');
+      this.args.uniqueid = args[0]; // if uniqueid is specified, it is always the first argument
+      this.args.mention.str = args[1]; // if user mention is specified, it is always the second argument
+    }
+  }
+  
+  
+  checkAuthenticity(){
+    // fetch validation variables
+    var globvar = globalVariables();
+    var teamid_true =  globvar['TEAM_ID'];
+    var token_true = PropertiesService.getScriptProperties().getProperty('VERIFICATION_TOKEN'); // expected verification token that accompanies slack API request     
+  
+    // initialise return message
+    var output = {code:false, msg:''};
+    
+    // check token
+    if(!token_true){ // check that token_true has been set in script properties
+      output.msg = 'error: VERIFICATION_TOKEN is not set in script properties. The command will not run. Please contact the web app developer.';
+      return output;
+    }  
+    if (this.token !== token_true) {
+      output.msg = 'error: Invalid token ' + this.token + ' . The command will not run. Please contact the web app developer.';
+      return output;
+    }
+    
+    // check request originates from our slack workspace
+    if (this.teamid != teamid_true){
+      output.msg = 'error: You are sending your command from an unauthorised slack workspace.';
+      return output;
+    }
+    
+    output.code=true;
+    return output;
+  }
+  
+  
+  slackCmd2FctName(){
+    // match this.subtype with the function it is meant to call
+    var slackCmd2FctName = globalVariables()['SLACKCMD_TO_FUNCTIONNAME']; // this is where all the key-value pairs (event.subtype - functionName) are stored
+    if (!slackCmd2FctName.hasOwnProperty(this.subtype)){ // if no key is found for this.subtype, return error
+        return false;
+    }
+    return slackCmd2FctName[this.subtype];
+  }
+  
+  
+  checkArgSyntax(){
+    // check syntax of this.args depending on function to be called
+    
+    // match doPost event.subtype with the function it is meant to call
+    var fctName = this.slackCmd2FctName();
+    if (!fctName){
+      return ('error: Sorry, the `' + this.subtype + '` command is not currently supported.');;
```

Comment:
Similar to my previous comment, no need to changing things now, but going forward, it would be cleaner to use string literals for things like this: https://developers.google.com/apps-script/guides/v8-runtime#template_literals